tsconfig.json

```diff
@@ -18,10 +18,11 @@
     "sourceMap": true,                     /* Generates corresponding '.map' file. */
 
     /* Strict Type-Checking Options */
-    "alwaysStrict": true,                   /* Parse in strict mode and emit "use strict" for each source file. */
+    "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
 
     /* Module Resolution Options */
-    "moduleResolution": "node",               /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
-    "types": ["node"], /* Type declaration files to be included in compilation. */
-  }
+    "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    "types": ["node"]                      /* Type declaration files to be included in compilation. */
+  },
+  "include": ["auth/**/*", "cloudant/**/*", "lib/**/*", "test/examples/src/ts/*"]
```

Comment:
I think I'd prefer if we did `"include": "**"` (not sure we actually need to have the line as it is the default) and then added `exclude` (similar to the `.eslintignore`) because I think it will be easier for us to maintain. In the `exclude` based approach when we add new `ts` it will be  transpiled rather than ignored, but we still get the benefits of reducing the amount of content that needs scanning to find `ts` files.

Reply:
I am against this idea.
The exclude can have leaks, so this opens up the possibility that eslint will be slow again, anytime, if handled with  white list then we can guarantee what is included in TS linter.
WDYT about only include files with `.ts` extension: 2d084a8 ?
This way the execution time takes a little bit more time, but not disturbing and also requires no config maintain in the future.

Reply:
>WDYT about only include files with `.ts` extension: 2d084a8 ? This way the execution time takes a little bit more time, but not disturbing and also requires no config maintain in the future.

This sounds ideal, but I am struggling to see why it is necessary because the [tsonfig docs](https://www.typescriptlang.org/tsconfig#include) say the default for `include` is:
>Default: [] if `files` is specified, `**` otherwise. 

_and_

>If a glob pattern doesn’t include a file extension, then only files with supported extensions are included (e.g. .ts, .tsx, and .d.ts by default, with .js and .jsx if allowJs is set to true).

they also say the default for `exclude` is:
>Default: `node_modules` `bower_components` `jspm_packages`

So my reading is that the default is effectively `**/*.ts` (plus 2 other `ts` related extensions) ignoring everything in `node_modules` - so I don't really understand why we had a problem in the first place.

I tried out locally and this is what I observed:

| Include pattern | With a cache | No cache yet |
| --- | --- | --- |
none/default | 1.69 s | 8.02 s
`"./**/*.ts"` | 1.65 s | 8.39 s

Reply:
Cool! I've not noticed this! Okay, at the beginning it was extremely slow, you shall compare it with the master run.
Removed this file filter: c1400aa